### DISCUSSION TITLE
Add bottom-sheet property to QPopupProxy

### DISCUSSION THREAD

**Discussion by lugrinder** [quasarframework/quasar#12551] [Ideas / Proposals]
Add a new property `bottom-sheet` to QPopupProxy to render the menu on smaller screens as BottomSheet plugin.
BottomSheet plugin is a QDialog with special settings. User can set up the QDialog of QPopupProxy to have the same appearance, but this will affect when rendered as QMenu on larger screens.
Another solution is to pass a property to the slot indicating the component used by proxy, and then the user can customize the content.

**Comment by pdanpdan** [comment]
You mean like this?
https://codepen.io/pdanpdan/pen/NWwYQNv?editors=1000

**Comment by lugrinder** [comment]
Yes, I'm doing this now, but instead of set breakpoint as a fixed value like in the example (800px), I'm using $q.screen.sizes.sm, but this method is too verbose. You need to set many classes and add components like QCard not needed by QMenu when using QList for the menu, but necessary for QDialog.
This idea or proposal is to simplify the recreation of the BottomSheet behavior and ensure that will render the same and get a unified UI.
